sentry/src/test/java/io/sentry/util/SentryRandomTest.kt

```diff
@@ -0,0 +1,45 @@
+package io.sentry.util
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+class SentryRandomTest {
+
+    @Test
+    fun `thread local creates a new instance per thread but keeps re-using it for the same thread`() {
+        val mainThreadRandom1 = SentryRandom.current()
+        val mainThreadRandom2 = SentryRandom.current()
+        assertEquals(mainThreadRandom1.toString(), mainThreadRandom2.toString())
```

Comment:
We could also use `assertSame`, as it's checking for the same reference:
```suggestion
        assertSame(mainThreadRandom1, mainThreadRandom2)
```